lib/wasix/src/syscalls/journal/mod.rs

```diff
@@ -31,9 +31,9 @@ pub struct JournalSyscallPlayer<'a, 'c> {
     pub ctx: FunctionEnvMut<'c, WasiEnv>,
     pub bootstrapping: bool,
 
-    pub journal_module_hash: Option<[u8; 8]>,
+    pub journal_module_hash: Option<Vec<u8>>,
     pub rewind: Option<RewindState>,
-    pub cur_module_hash: [u8; 8],
+    pub cur_module_hash: Vec<u8>,
```

Comment:
We can probably use a Boxed slice instead, as we'll need no conversion? (`Box<[u8]>`)